server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -222,7 +228,8 @@ public IndexService(
         BiFunction<IndexSettings, ShardRouting, TranslogFactory> translogFactorySupplier,
         Supplier<TimeValue> clusterDefaultRefreshIntervalSupplier,
         RecoverySettings recoverySettings,
-        RemoteStoreSettings remoteStoreSettings
+        RemoteStoreSettings remoteStoreSettings,
+        FileCache fileCache
```

Comment:
@rayshrey do we need to amend existing ctors or we can use another ctors for this? Checking we can minimize the blast radius.

Reply:
Agree. This also breaks the Public API contract for these APIs.
Have created a new const to handle this issue in follow up revision